### PR TITLE
[Core] Adding wrapper to mortar mapper

### PR DESCRIPTION
**Pull request by loumalouomega**
Additionally adding mapping coefficient

### CODE REVIEW

**Comment by philbucher** [comment]
for the future I think it would be good if the interface would follow the Mapper-Interface in the MappingApp

**Comment by rubenzorrilla** [comment]
Actually I don't know which interface is better, but I agree on having the same interface for all the mappers in Kratos.

**Comment by philbucher** [comment]
The interface in the MappingApp is what we agreed on (pretty much everyone even remotely involved with mapping) when I was in Barcelona two years ago

I still like it, but you are welcome to join the discussion in #3108 where this is currently under discussion

also it is documented in the wiki for a long time now: https://github.com/KratosMultiphysics/Kratos/wiki/How-to-do-Mapping-between-nonmatching-meshes

**Comment by loumalouomega** [comment]
The constructor are the same, I can do the same interface adding some custom methods

**Comment by loumalouomega** [comment]
For a future PR

**Comment by loumalouomega** [comment]
Can anyone approve this?

**Comment by rubenzorrilla** [comment]
i find this wrapper idea a bit cumbersome but honestly I can't figure out a better way to do it. In any case, I would prefer to discuss it further.

**Comment by philbucher** [comment]
I am fine with the changes
In my implementations the user also does not have to specify the geometry

**Comment by loumalouomega** [comment]
@philbucher I added a Map method, as an experiment, I tried the InverseMap too, but I couldn't because there is no swap method in the ModelPart. Additionally mention that I can not add the Map method to the Wrapper, due to the templates related with the variable type. I need to think about this

**Comment by loumalouomega** [comment]
Besides I merged this with #3479

**Comment by loumalouomega** [comment]
Ping

**Comment by loumalouomega** [comment]
Ping

**Comment by loumalouomega** [comment]
Ping @rubenzorrilla

**Comment by philbucher** [comment]
can you merge master?
I think there are some unrelated/already merged changes

**Comment by loumalouomega** [comment]
> can you merge master?
> I think there are some unrelated/already merged changes

I need approval

**Comment by philbucher** [comment]
I mean can you merge master into this branch?

**Comment by loumalouomega** [comment]
> I mean can you merge master into this branch?

I think so

**Comment by philbucher** [comment]
> > I mean can you merge master into this branch?
> 
> I think so

I will take another look after you did this :)

**Comment by loumalouomega** [comment]
> > > I mean can you merge master into this branch?
> > 
> > 
> > I think so
> 
> I will take another look after you did this :)

Yes, I could

**Comment by loumalouomega** [comment]
Ping

File: kratos/processes/simple_mortar_mapper_process.cpp [hunk 1]
```diff
@@ -89,6 +118,25 @@ void SimpleMortarMapperProcess<TDim, TNumNodes, TVarType, TNumNodesMaster>:: Exe
...
+            #pragma omp parallel for
+            for (int k = 0; k< static_cast<int> (r_nodes_array.size()); ++k) {
+                auto it_node = r_nodes_array.begin() + k;
+                it_node->FastGetSolutionStepValue(mDestinationVariable) *= mMappingCoefficient;
+            }
+        } else {
+            #pragma omp parallel for
+            for (int k = 0; k< static_cast<int> (r_nodes_array.size()); ++k) {
+                auto it_node = r_nodes_array.begin() + k;
+                it_node->GetValue(mDestinationVariable) *= mMappingCoefficient;
```
Comment (RiccardoRossi) [inline kratos/processes/simple_mortar_mapper_process.cpp:135]: are u sure that mDestinationVariable exists?

**Review by RiccardoRossi — COMMENTED** [review]
I am ok with this, nevertheless this is on @rubenzorrilla and @philbucher to accept

File: kratos/processes/simple_mortar_mapper_process.cpp (hunk 1 above)
Comment (loumalouomega) [reply kratos/processes/simple_mortar_mapper_process.cpp:135]: Yes, it is set previously

**Review by rubenzorrilla — APPROVED** [review]
Go ahead!

**Comment by loumalouomega** [comment]
Thanks

**Pull request merged**